contracts/exchange/ibetEscrow.sol

```diff
@@ -0,0 +1,422 @@
+    /// @param _token トークンアドレス
+    /// @return 残高数量
+    function depositOf(address _account, address _token)
+        public
+        view
+        returns (uint256)
+    {
+        return EscrowStorage(storageAddress).getDeposit(
+            _account,
+            _token
+        );
+    }
+
+    // ---------------------------------------------------------------
+    // Function: Logic
+    // ---------------------------------------------------------------
+
+    /// @notice エスクロー新規作成
+    /// @param _token トークンアドレス
+    /// @param _recipient トークン受領者
+    /// @param _amount 数量
+    /// @param _agent エスクローエージェント
+    /// @param _data イベント出力用の任意のデータ
+    function createEscrow(
+        address _token,
+        address _recipient,
+        uint256 _amount,
+        address _agent,
+        string memory _data
+    )
+        public
+        returns (bool)
+    {
+        // チェック：数量がゼロより大きいこと
+        require(
+            _amount > 0,
+            "The amount must be greater than zero."
+        );
+
+        // チェック：数量が残高以下であること
+        require(
+            balanceOf(msg.sender, _token) >= _amount,
+            "The amount must be less than or equal to the balance."
+        );
+
+        // チェック：トークンのステータスが有効であること
+        require(
+            IbetStandardTokenInterface(_token).status() == true,
+            "The status of the token must be true."
+        );
+
+        // 更新：エスクローIDをカウントアップ
+        uint256 _escrowId = EscrowStorage(storageAddress).getLatestEscrowId() + 1;
+        EscrowStorage(storageAddress).setLatestEscrowId(_escrowId);
+
+        // 更新：エスクロー情報の挿入
+        EscrowStorage(storageAddress).setEscrow(
+            _escrowId,
+            _token,
+            msg.sender,
+            _recipient,
+            _amount,
+            _agent,
+            false
+        );
+
+        // 更新：残高
+        setBalance(msg.sender, _token, balanceOf(msg.sender, _token).sub(_amount));
+
+        // 更新：エスクロー中数量
+        setDeposit(msg.sender, _token, depositOf(msg.sender, _token).add(_amount));
+
+        // イベント登録
+        emit EscrowCreated(
+            _escrowId,
+            _token,
+            msg.sender,
+            _recipient,
+            _amount,
+            _agent,
+            _data
+        );
+
+        return true;
+    }
+
+    /// @notice エスクロー取消
+    /// @param _escrowId エスクローID
+    function cancelEscrow(uint256 _escrowId)
+        public
+        returns (bool)
+    {
+        // チェック：エスクローIDが直近ID以下であること
+        require(
+            _escrowId <= EscrowStorage(storageAddress).getLatestEscrowId(),
+            "The escrowId must be less than or equal to the latest escrow ID."
+        );
+
+        Escrow memory escrow;
+        (escrow.token, escrow.sender, escrow.recipient,
+            escrow.amount, escrow.agent, escrow.status) =
+                EscrowStorage(storageAddress).getEscrow(_escrowId);
+
+        // チェック：エスクローが有効であること
+        require(
+            escrow.status == false,
```

Comment:
[IMO]`status` may be difficult to understand...

Reply:
You're right, and I did miss the logic.
I will fix it.